sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsAtomicPartitionManagement.java

```diff
@@ -50,11 +49,11 @@ public interface SupportsAtomicPartitionManagement extends SupportsPartitionMana
   default void createPartition(
       InternalRow ident,
       Map<String, String> properties)
-      throws PartitionAlreadyExistsException, UnsupportedOperationException {
+      throws PartitionsAlreadyExistException, UnsupportedOperationException {
```

Comment:
Does Spark try-catch the errors thrown by these 2 APIs? If yes then we may have bugs as the existing data source implementations may still throw `PartitionAlreadyExistsException`

Reply:
In the current implementation, Spark invokes `createPartition()`/`createPartitions()` **only** from `AddPartitionExec` (ALTER TABLE .. ADD PARTITION) where all exceptions from the methods are propagated to users directly. **But it doesn't matter** in our case because `AddPartitionExec.run()` checks partitions exist BEFORE the invokes, see:
https://github.com/apache/spark/blob/5d74ace648422e7a9bff7774ac266372934023b9/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AddPartitionExec.scala#L41-L47

Reply:
I think it's OK to change the type of the exception that is propagated to users.